misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/CodeGenerator.java

```diff
@@ -273,22 +273,25 @@ private Template compileTemplate(String defaultTemplateDir, String templateName)
         handlebars.setInfiniteLoops(true); //This will allow templates to call themselves with recursion.
         handlebars.registerHelpers(StringHelpers.class);
         handlebars.registerHelper("equals", (object, options) -> {
-            CharSequence result;
+            CharSequence result = null;
             Object param0 = options.param(0);
-
-            if (param0 == null) {
-                throw new IllegalArgumentException("found 'null', expected 'string'");
-            }
-            if (object != null) {
-                if (object.toString().equals(param0.toString())) {
-                    result = options.fn(options.context);
+            try {
+                if (param0 == null) {
```

Comment:
I guess the null should be handled from the model instead of the template helper